partition/bootloader.go

```diff
@@ -146,3 +155,13 @@ func MarkBootSuccessful(bootloader Bootloader) error {
 
 	return bootloader.SetBootVars(m)
 }
+
+func reboot() error {
```

Comment:
By mocking the actual exec call :-)

We do this elsewhere in the snapd code base already. https://github.com/snapcore/snapd/blob/master/systemd/systemd_test.go is a good reference.